src/mds/MDCache.cc

```diff
@@ -1414,6 +1425,47 @@ void MDCache::adjust_subtree_after_rename(CInode *diri, CDir *olddir, bool pop)
   show_subtrees();
 }
 
+// ===================================
+// Default cache memory limit
+
+
+/*
+ * Get the default cache memory limit using the mds_memory_target
+ */
+uint64_t MDCache::get_default_mds_cache_memory_limit()
+{
+  uint64_t temp_cache_limit = (uint64_t) (mds->memory_target/1.25);
+  mds->cct->_conf.set_val_default("mds_cache_memory_limit", stringify(temp_cache_limit));
+  return temp_cache_limit;
+}
+
+// ===================================
+// Cache Autotuning
+
+
+/*
+ * If mds_cache_autotune is set to true, then adjust cache_memory_limit using the rss usage periodically as per the mds_cache_resize_interval
+ */
+void MDCache::adjust_cache_memory_limit()
```

Comment:
Let's rename this `MDCache::resize` and call it from https://github.com/ceph/ceph/blob/f590518a2f0962fa39604531eb4841cc1ff13f8e/src/mds/MDSRank.cc#L746